core/src/stack/inputs.rs

```diff
@@ -94,27 +92,17 @@ impl ToElements<Felt> for StackInputs {
 
 impl Serializable for StackInputs {
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
-        // TODO the length of the stack, by design, will not be greater than `u32::MAX`. however,
-        // we must define a common serialization format as we might diverge from the implementation
-        // here and the one provided by default from winterfell.
-
-        debug_assert!(self.values.len() <= Self::MAX_LEN);
-        target.write_usize(self.values.len());
-        target.write_many(&self.values);
+        debug_assert!(self.values.len() == STACK_DEPTH);
+        target.write_many(self.values);
```

Comment:
nit: this always serializes exactly 16 values - should we skip serializing any trailing zero values? That is, let's say that the 3rd element on the stack is non-zero, and all remaining elements are zeros. Then, we can serialize only the first 4 elements.